---
---
<!DOCTYPE html>
<html>
{% include header.html %}
  <body>

    <header>
      <div class="container">
        <h1>Pybamview</h1>
        <h2>Browser-based application for viewing bam alignments</h2>
	{% include menu.html %}<br>
      </div>
    </header>
    <div class="container">
      <section id="main_content">
      <h1>snapbam: automatically generate PyBamView snapshots</h1>
      This page describes how to make alignment snapshots on the command line. It assumes you have already gone through the <a href="install.html">install</a> instructions 

      <h2>example command</h2>
      To get started with files from the <a href="http://www.broadinstitute.org/~gymrek/pybamview_example_data.tar.gz">examples download</a>:
        {% highlight bash %}
    snapbam \
    --bam examples/example.sorted.bam \
    --regions examples/example_targets.bed  \
    --zoom 1 \
    --samples mysample \
    --outdir . \
    --filetype html \
    --buffer 100
    {% endhighlight %}
    This will generate html files for each locus in the current directory, e.g. "PentaE.html".

      <h2>full command line usage</h2>
To see the usage information for snapbam, type:
{% highlight bash %}
>snapbam --help
Usage: snapbam [options]

Automated generation of pybamview snapshots

Options:
  -h, --help            show this help message and exit
  --bam=BAM             Run Pybamview on this BAM file only. The file must be
                        indexed.
  --bamdir=BAMDIR       Directory to look for bam files. Bam files must be
                        indexed.
  --ref=REF             Path to reference fasta file. If no reference is
                        given, the reference track is displayed as "N"s
  --regions=REGIONS     Bed file with alignment regions to display. Columns:
                        chrom, start, end, name
  --zoom=ZOOM           Zoom level (Options: 1-10, 50, 100)
  --samples=SAMPLES     Comma-separated list of samples to display
  --outdir=OUTDIR       Directory to write output
  --buffer=BUFFER       Include this many nucleotides buffer upstream and
                        downstream. Between 0-500.
  --filetype=FILETYPE   Filetype of snapshots. Options: svg, pdf, html
  --downsample=DOWNSAMPLE
                        Downsample reads to this maximum coverage level.
                        Default: 50. Values over 100 not recommended since it
                        results in poor performance.
{% endhighlight %}
      </section>
    </div>
    {% include footer.html %}
  </body>
</html>
